plenum/server/replica.py

```diff
@@ -335,82 +418,108 @@ def viewNo(self):
         """
         return self.node.viewNo
 
-    def isPrimaryInView(self, viewNo: int) -> Optional[bool]:
-        """
-        Return whether a primary has been selected for this view number.
-        """
-        return self.primaryNames[viewNo] == self.name
-
-    def isMsgForLaterView(self, msg):
-        """
-        Return whether this request's view number is greater than the current
-        view number of this replica.
-        """
-        viewNo = getattr(msg, "viewNo", None)
-        return viewNo > self.viewNo
-
-    def isMsgForCurrentView(self, msg):
-        """
-        Return whether this request's view number is equal to the current view
-        number of this replica.
-        """
-        viewNo = getattr(msg, "viewNo", None)
-        return viewNo == self.viewNo
-
-    def isMsgForPrevView(self, msg):
-        """
-        Return whether this request's view number is less than the current view
-        number of this replica.
-        """
-        viewNo = getattr(msg, "viewNo", None)
-        return viewNo < self.viewNo
-
-    def isPrimaryForMsg(self, msg) -> Optional[bool]:
-        """
-        Return whether this replica is primary if the request's view number is
-        equal this replica's view number and primary has been selected for
-        the current view.
-        Return None otherwise.
-
-        :param msg: message
-        """
-        if self.isMsgForLaterView(msg):
-            self.discard(msg,
-                         "Cannot get primary status for a request for a later "
-                         "view {}. Request is {}".format(self.viewNo, msg),
-                         logger.error)
-        else:
-            return self.isPrimary if self.isMsgForCurrentView(msg) \
-                else self.isPrimaryInView(msg.viewNo)
-
     def isMsgFromPrimary(self, msg, sender: str) -> bool:
         """
         Return whether this message was from primary replica
         :param msg:
         :param sender:
         :return:
         """
-        if self.isMsgForLaterView(msg):
-            logger.error("{} cannot get primary for a request for a later "
-                         "view. Request is {}".format(self, msg))
-        else:
-            return self.primaryName == sender if self.isMsgForCurrentView(
-                msg) else self.primaryNames[msg.viewNo] == sender
+        # if self.isMsgForLaterView(msg):
+        #     logger.error("{} cannot get primary for a request for a later "
+        #                  "view. Request is {}".format(self, msg))
+        # else:
+        # return self.primaryName == sender if self.isMsgForCurrentView(
+        #     msg) else self.primaryNames[msg.viewNo] == sender
+        return self.primaryName == sender
```

Comment:
Initially the replica used to handle previous and future view messages, not it discards previous view messages and node handles future view messages